core/src/transaction_id.rs

```diff
@@ -0,0 +1,78 @@
+use serde::{Deserialize, Serialize};
+use std::{borrow::Cow, fmt, str};
+
+/// transaction identifier
+///
+/// needs to be in a human readable format. Usually this is going to be
+/// in hexadecimal. However this is not necessarily guaranteed. Knowing
+/// exactly the formatting is not necessary for what we intend to do any
+/// way.
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Hash, Clone, Serialize, Deserialize)]
+pub struct TransactionId(Cow<'static, str>);
```

Comment:
it's a bit off-topic, because I know this is already done this way, but I've been wondering if we could find a way of re-using the code between this and `BlockId`, for example. Maybe later though.

Reply:
we can think of sth like type alias for that

Reply:
if you use Type alias you will loose on the type system to drive your work. What you can do is use `Deref` to access the inner content's member functions without breaking the type abstraction.

Reply:
Or use a tagged type, like the `Value` in milkomeda validator. Although it's a bit leaky.

Or a macro. 

There are other types that are mostly the same thing in milkomeda-validator, not only these two, `Address` is mostly the same, `PolicyId`, etc... Maybe there is not a nice abstraction for everything though, but I feel like there may be some abstraction missing somewhere.